master/pkg/schemas/expconf/hyperparameters_config.go

```diff
@@ -22,6 +47,7 @@ type HyperparameterV0 struct {
 	RawDoubleHyperparameter      *DoubleHyperparameterV0      `union:"type,double" json:"-"`
 	RawLogHyperparameter         *LogHyperparameterV0         `union:"type,log" json:"-"`
 	RawCategoricalHyperparameter *CategoricalHyperparameterV0 `union:"type,categorical" json:"-"`
+	RawNestedHyperparameter      *map[string]HyperparameterV0 `union:"type,object" json:"-"`
```

Comment:
The `union:"type,object"` thing here would indicate that it was possible to have a map of hyperparameters like:
```yaml
hyperparameters:
  global_batch_size: 8
  nest:
    type: object
    value:
      a: 1
      b: 2
```

It might not be bad to allow that, but right now we don't.

Using the union package this way is a bit of a hack though, since you are including the union tag to benefit from some behaviors but you are manually avoiding other behaviors of the union tag (marshaling and unmarshaling).

I am actually ok with the way it is but it would be nice to include a comment above `RawNestedHyperparameter` that explains why you are assigning it a `union` tag even though you never want it to marshal or unmarshal with the usual union logic.